olive/passes/onnx/genai_model_exporter.py

```diff
@@ -69,25 +81,37 @@ def is_accelerator_agnostic(accelerator_spec: AcceleratorSpec) -> bool:
         return False
 
     def _run_for_config(
-        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+        self,
+        model: Union[PyTorchModelHandler, ONNXModelHandler],
+        data_root: str,
+        config: Dict[str, Any],
+        output_model_path: str,
     ) -> ONNXModelHandler:
         from onnxruntime_genai.models.builder import create_model
 
-        if not model.hf_config:
+        precision = config["precision"]
+        metadata_only = config["metadata_only"]
+
+        if not metadata_only and not model.hf_config:
             raise ValueError(
-                "GenAIModelExporter pass only supports exporting HF models i.e. PyTorchModelHandler with hf_config."
+                "GenAIModelExporter pass only supports exporting HF models i.e. PyTorchModelHandler "
+                "with hf_config and exporting the metadata only for pre-optimized onnx models."
             )
 
         Path(output_model_path).mkdir(parents=True, exist_ok=True)
-        output_model_filepath = Path(resolve_onnx_path(output_model_path))
+        output_model_filepath = (
+            Path(resolve_onnx_path(output_model_path))
+            if isinstance(model, PyTorchModelHandler)
+            else Path(resolve_onnx_path(output_model_path, model.onnx_file_name))
```

Comment:
I am a bit unsure about the else part. 
With this output_model_filepath, the metadata will be saved in a fresh directory. but the actual `model.onnx` and `model.onnx.data` files are in the model.get_resource("model_path") directory. 

since the input model is copied in this case, during packages the metadata would be missing since they are not in the model_path folder.

Reply:
I think our options are
- copy over preexisting files into the new `output_model_path` and create a new onnx model handler with this path
- generate the metadata directly into the input model's directory and return a copied handler with the additional file attributes.

Reply:
I don't think it does? 
Since the genai_config.json is created inside a different folder than the "model_path" resource of the output_model, I think the packaged model will be missing this file.

Reply:
I verified both paths (Pytorch model as input + GenAIExporter and OnnxConversion + GenAIExporter with metadata). Both succeeded with all the necessary files in the packaged zip file.

Reply:
The current changes to cache.py were not present when I made this comment (maybe got lost in the rebase and force push).

With the changes it seems okay now. But I think we will probably have issues in the future if we want to use the generative api directly on the cached model due to the model.onnx and genai_config.json files being in different folders in cache (you could verify this by checking your Olive cache)

But that can be fixed later. 

I think we have to make some decisions on whether the extra file paths should be members of the model_path directory or they can be under other directories. But that can wait after this PR.

Reply:
Yes, evaluation from cache folder will be an issue. In my opinion, the right solution here is to use symlinks and each pass should have ALL its output under one root folder. Passes like the ORT perf tuning generate no output of its own but provides the previous pass's output as the output folder. That's kind of hack. If we had symlinks then the output folder can still be its own with symlinks into the previous passes output folder.

Sure deserves a revisit.